### ISSUE TITLE
spurious error with cyclic imports and NamedTuple

### ISSUE THREAD

**Issue by rwbarton** [python/mypy#1567]
File `nt.py`:
```py
from typing import NamedTuple

import ntb

class A(NamedTuple('A', [('x', int)])): pass

class B:
    def __init__(self):
        self.a = None  # type: A
```
File `ntb.py`:
```py
from typing import Callable

from nt import A, B

def g(f: Callable[[A], None], b: B) -> None:
    f(b.a)
```

`mypy ntb.py` works okay, but `mypy nt.py` gives the following error:
```
nt.py:3: note: In module imported here:
ntb.py: note: In function "g":
ntb.py:6: error: Argument 1 has incompatible type "A"; expected "nt.A"
```
Although the error formatting obscures it, the types involved are really `Tuple[builtins.int, fallback=nt.A]` and `nt.A` respectively. This only happens when `A` is a named tuple class.

**Comment by gvanrossum** [comment]
Looks like this was fixed in 0.4.3 (I don't care enough to bisect down to the exact rev.)

**Comment by kirbyfan64** [comment]
@gvanrossum Fixed by 9a92899e7c8416d43196d8bf424e7e2d86f27f0d.

**Comment by gvanrossum** [comment]
Thanks for the bisect! :-)
